### PR TITLE
Add PageFactory to amazon.com test

### PR DESCRIPTION
**Pull request by emmaaghlamazyan94**

### CODE REVIEW

File: src/main/java/pages/BooksPage.java [hunk 1]
```diff
@@ -36,7 +44,7 @@ public void clickOnSearchResult() {
     public void waitUntilPageLoads() {
         try {
             wait.until(ExpectedConditions.numberOfElementsToBeMoreThan
-                    (By.xpath("//div[@class='s-include-content-margin s-border-bottom s-latency-cf-section']"), 0));
+                    ((By) books, 0));
```
Comment (David91) [inline src/main/java/pages/BooksPage.java:47]: `(By) books` conversion is impossible, as these types are incompatible. Your code always throws an exception at runtime and it's also unclear why you handle the exception when it's mindless to continue execution if a page isn't loaded.
Why you don't receive an error message compile-time? The reason perhaps is hidden in a fact, that `books` is a WebElement actually, which is an interface and the compiler isn't able to detect the actual object type on which books variable refers to. It is always known at runtime.
So remove this cast and don't handle the exception thrown by the explicit wait, as it's mindless to continue if the page isn't loaded.

File: src/main/java/pages/AuthorPage.java [hunk 2]
```diff
@@ -26,12 +26,23 @@
...
         this.driver = driver;
         this.wait = new WebDriverWait(driver, 10);
         PageFactory.initElements(driver, this);
     }
 
+    public List<WebElement> authorsList() {
+        List<WebElement> authors = byAuthors;
+        return authors;
+    }
+
```
Comment (David91) [inline src/main/java/pages/AuthorPage.java:45]: What's the meaning of this method? Not counting that method returning `List<WebElement>` shouldn't be public, as it's against Page Object principles.

**Review by David91 — CHANGES_REQUESTED** [review]

File: src/main/java/pages/AuthorPage.java (hunk 2 above)
Comment (emmaaghlamazyan94) [reply src/main/java/pages/AuthorPage.java:45]: Capture
I wanted to fix this comment

Comment (emmaaghlamazyan94) [reply src/main/java/pages/AuthorPage.java:45]: Got List of authors from page, and wrote logic in Test class

File: src/main/java/pages/BooksPage.java [hunk 3]
```diff
@@ -34,19 +36,25 @@ public String getActualText() {
...
-            System.out.println("No results found in Books.");
+    public boolean checkAllBooksBySameAuthor(String authorFullName) {
+        List<WebElement> authorsList = authors;
+        for (WebElement author : authorsList) {
+            if (!author.getText().toLowerCase().contains(authorFullName)) {
+                return false;
+            }
         }
+        return true;
+    }
```
Comment (David91) [inline src/main/java/pages/BooksPage.java:54]: Remind me tomorrow to talk about this kind of method.

**Review by David91 — APPROVED** [review]

**Pull request merged**